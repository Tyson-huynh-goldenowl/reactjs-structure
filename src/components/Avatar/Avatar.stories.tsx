import React from "react";
import { Story, Meta } from "@storybook/react";
import { UserOutlined, AntDesignOutlined } from '@ant-design/icons';
import { Avatar } from "../../components";
import { AvatarProps } from "./Avatar";

export default {
  title: "Example/Avartar",
  component: Avatar,
} as Meta;

const Template: Story<AvatarProps> = (args) => <Avatar {...args} />;


export const Large = Template.bind({});
Large.args = {
  size: 40,
  icon: <UserOutlined />
};

export const Small = Template.bind({});
Small.args = {
  size: 80,
  icon: <AntDesignOutlined />
};
