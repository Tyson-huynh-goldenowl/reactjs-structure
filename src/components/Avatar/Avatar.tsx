import React from "react";
import { Avatar } from 'antd';

export interface AvatarProps {
  size?: number;
  icon?: any;
}

const AvatarComponent = ({ size, icon, ...props }: any) => {
  return <Avatar
    icon={icon}
    size={size || 40}
  />
};

export default AvatarComponent;
