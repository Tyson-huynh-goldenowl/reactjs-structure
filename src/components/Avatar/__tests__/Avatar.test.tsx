import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { UserOutlined } from '@ant-design/icons';
import Avatar from "../Avatar";

describe('Avatar component', () => {
  it('renders correctly', () => {

    const output = shallow(
      <Avatar
        icon={<UserOutlined />}
        size={80}
      />
    );
    expect(shallowToJson(output)).toMatchSnapshot();
    expect(shallow(<Avatar icon={<UserOutlined />} size={80} />).contains(<Avatar icon={<UserOutlined />} size={80} />)).toBe(false);

  });

})