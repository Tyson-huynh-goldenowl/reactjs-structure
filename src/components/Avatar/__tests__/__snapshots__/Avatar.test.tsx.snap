// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar component renders correctly 1`] = `
<Avatar
  icon={<ForwardRef(UserOutlined) />}
  shape="circle"
  size={80}
/>
`;
